src/Microsoft.ML.Data/Transforms/ChooseColumnsTransform.cs

```diff
@@ -442,6 +442,18 @@ private static VersionInfo GetVersionInfo()
 
         private const string RegistrationName = "ChooseColumns";
 
+        /// <summary>
+        /// Convenience constructor for public facing API.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
+        /// <param name="name">Name of the output column.</param>
+        /// <param name="source">Name of the selected column.  If this is null '<paramref name="name"/>' will be used.</param>
+        public ChooseColumnsTransform(IHostEnvironment env, IDataView input, string name, string source = null)
```

Comment:
Choose column transform is one of those transforms that is pretty much completely useless unless it supports multiple columns. It's not like other transforms where you can just apply the transform again to get more columns. If something isn't in the choose list, it is simply dropped -- no second chances. We need to support multiple column specification in the constructor.

However, choose columns *generally* as far as I know could just do with a `params string[] name` or something, and be done with it. Generally it isn't used to rename columns. (That is, I see `xf=choose{col=Foo,Bar,Biz}`, I don't see much `xf=Choose{col=Foo:Bar}` or something like this.)

Reply:
Actually, renaming of chosen column is the difference between ChooseColumnsTransform and KeepColumnsTransform. KeepColumnsTransform does not support renaming and drops the rest of the columns. ChooseColumnsTransform does renaming as well as  has other parameters like `Hidden` which has following options.

``` C#
public enum HiddenColumnOption : byte
{
    Drop = 1,
    Keep = 2,
    Rename = 3
}
```

What I suggest that for convenience this transform operate in similar way as KeepColumnsTransform. 
``` C#
 public ChooseColumnsTransform(IHostEnvironment env, IDataView input, param string[] columnNames)
            : this(env, new Arguments(columnNames) {  }, input)
{
}
```

For other usage, user can invoke constructor with `Arguments` object.

Reply:
I'd probably just keep the non-renaming case convenient, and nuke the constructor on arguments.

---
In reply to: [200471726](https://github.com/dotnet/machinelearning/pull/491#discussion_r200471726) [](ancestors = 200471726)

Reply:
Do you mean that current changes are fine? If not can you please explain bit more?

---
In reply to: [200727095](https://github.com/dotnet/machinelearning/pull/491#discussion_r200727095) [](ancestors = 200727095,200471726)

Reply:
They are good, thank you @zeahmed.

---
In reply to: [200743746](https://github.com/dotnet/machinelearning/pull/491#discussion_r200743746) [](ancestors = 200743746,200727095,200471726)